flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/DeclarativeSlotManager.java

```diff
@@ -717,6 +717,12 @@ public ResourceProfile getFreeResourceOf(InstanceID instanceID) {
         return taskExecutorManager.getTotalFreeResourcesOf(instanceID);
     }
 
+    @Override
+    public Collection<TaskManagerSlotInformation> getAllocatedSlotsOf(InstanceID instanceID) {
+        // This information is only valid for fine-grained resource management.
```

Comment:
I would say it's currently not supported for this slot manager. The information should always be valid.

Same for `SlotManagerImpl`.